src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests.vb

```diff
@@ -363,16 +365,114 @@ BC30581: 'AddressOf' expression cannot be converted to 'Integer' because 'Intege
+      IOperation:  (OperationKind.None, IsInvalid) (Syntax: 'AddressOf Main')
+        Children(1):
+            IOperation:  (OperationKind.None, IsInvalid) (Syntax: 'Main')
+              Children(1):
+                  IInstanceReferenceExpression (OperationKind.InstanceReferenceExpression, Type: Module1, IsInvalid) (Syntax: 'Main')")
 
             comp.VerifyOperationTree(nodes(3), expectedOperationTree:=
 "IInvalidExpression (OperationKind.InvalidExpression, Type: System.Void, IsInvalid) (Syntax: 'Test2(New S ... essOf Main)')
   Children(3):
       IOperation:  (OperationKind.None) (Syntax: 'Test2')
+        Children(1):
+            IInstanceReferenceExpression (OperationKind.InstanceReferenceExpression, Type: Module1) (Syntax: 'Test2')
       IObjectCreationExpression (Constructor: Sub System.Guid..ctor()) (OperationKind.ObjectCreationExpression, Type: System.Guid, IsInvalid) (Syntax: 'New System.Guid()')
         Arguments(0)
         Initializer: null
-      IOperation:  (OperationKind.None, IsInvalid) (Syntax: 'AddressOf Main')")
+      IOperation:  (OperationKind.None, IsInvalid) (Syntax: 'AddressOf Main')
+        Children(1):
+            IOperation:  (OperationKind.None, IsInvalid) (Syntax: 'Main')
+              Children(1):
+                  IInstanceReferenceExpression (OperationKind.InstanceReferenceExpression, Type: Module1, IsInvalid) (Syntax: 'Main')")
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact()>
+        Public Sub AddressOf_BoundMethod_ExposesReceiver()
+            Dim source = <![CDATA[
+Imports System
+Module Program
+    Sub Main(args As String())
+        Dim c1 As New C1
+        Console.WriteLine(New With {Key .a = AddressOf c1.S})'BIND:"New With {Key .a = AddressOf c1.S}"
+    End Sub
+
+    Class C1
+        Sub S()
+        End Sub
+    End Class
+End Module]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IAnonymousObjectCreationExpression (OperationKind.AnonymousObjectCreationExpression, Type: <anonymous type: Key a As ?>, IsInvalid) (Syntax: 'New With {K ... essOf c1.S}')
+  Initializers(1):
+      ISimpleAssignmentExpression (OperationKind.SimpleAssignmentExpression, Type: ?, IsInvalid) (Syntax: 'Key .a = AddressOf c1.S')
+        Left: IPropertyReferenceExpression: ReadOnly Property <anonymous type: Key a As ?>.a As ? (Static) (OperationKind.PropertyReferenceExpression, Type: ?) (Syntax: 'a')
+            Instance Receiver: null
+        Right: IInvalidExpression (OperationKind.InvalidExpression, Type: ?, IsInvalid) (Syntax: 'AddressOf c1.S')
+            Children(1):
+                IOperation:  (OperationKind.None, IsInvalid) (Syntax: 'AddressOf c1.S')
+                  Children(1):
+                      IOperation:  (OperationKind.None, IsInvalid) (Syntax: 'c1.S')
+                        Children(1):
+                            IOperation:  (OperationKind.None, IsInvalid) (Syntax: 'c1')
```

Comment:
Why are `c1` and `c1.S` invalid and with `OperationKind.None`?

Reply:
`c1.S` is the BoundMethodGroup here, so it's exposed as a NoneOperation. `c1` is actually a `BoundTypeOrValueExpression`, which we don't have support for.

Reply:
Why are they marked `IsInvalid`?

Reply:
Because the code doesn't compile. BoundAddressOf is not created unless the method fails to bind correctly.

Reply:
It looks like `BoundAddressOf` is created in this case, with `HasErrors = False`.

Reply:
Sure, but IsInvalid checks to see if the underlying syntax has diagnostics, which in this case it does. `BoundNode.HasErrors` is not used for this.